### PR TITLE
Streamline tests

### PR DESCRIPTION
**Pull request by khaeru**
**tl;dr:** this PR makes better use of pytest features instead of reinventing them in multiple places. It also moves some testing code to a new `ixmp.testing` submodule so it can be reused by message_ix (e.g. [this](https://github.com/iiasa/message_ix/blob/f80ad367c17c303512710d654c6743c8567e5028/tests/conftest.py#L19-L32
) shouldn't need redefinition) or by users.

I started making some of these changes when writing tests for #126 and figured they could stand alone.
- ixmp.testing: new submodule for reuse of testing utilities.
  - create_local_testdb(): moved from tests/testing_utils.py
  - dantzig_transport(): replaces code in tests/test_integration.py and
    tests/testdb_setup.py.
- consolidate test data in tests/data.
- ixmp.Platform.__init__: ensure dbprops is string, for Java compat.
- ixmp.default_paths: use Path objects.
- tests/conftest.py:
  - new fixtures: test_data_path, tutorial_path.
  - existing test_mp* fixtures: add docstrings, use tmp_path_factory
    fixture.
  - new hook: pytest_report_header (replaces test_admin.py).
- other tests: use pytest & ixmp fixtures.

**PR checklist:**

- [x] Tests added
- [x] Documentation added
- [x] Description in RELEASE_NOTES.md added

### CODE REVIEW

**Comment by khaeru** [comment]
Okay, finally all the tests pass. This is ready to review & merge, from my POV.

**Comment by khaeru** [comment]
> Would it be possible to connect the tutorial-notebook tests to a temporary db?
As mentioned in Slack, this is possible, but it requires new functionality. Related to #124. I would prefer to merge this PR and then work on the other issue.
